### PR TITLE
grpc-proto: modernize more for conan v2

### PR DESCRIPTION
**Pull request by SpaceIm**
- protect double delete of fPIC
- add layout
- fix access to dependencies options

other changes:
- sort methods by order of execution
- add required_conan_version
- trick to avoid protobuf in build requirement in case of native build
- remove protobuf full_package_mode (see https://github.com/conan-io/conan-center-index/pull/16018)

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

**Review by jwillikers — DISMISSED** [review]

**Review by prince-chrismc — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
Needed for https://github.com/conan-io/conan-center-index/pull/16564

**Review by SSE4 — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
Cla is stuck

**Comment by prince-chrismc** [comment]
#15601 #16034 ??

**Comment by SpaceIm** [comment]
> #15601 #16034 ??

They will generate missing v2 packages yes. I'll also bump protobuf in grpc-proto once googleapis v2 packages are available. Draft until packages are available.

**Comment by prince-chrismc** [comment]
#16034

File: recipes/grpc-proto/all/conanfile.py [hunk 1]
```diff
@@ -18,52 +23,70 @@ class GRPCProto(ConanFile):
...
+                self.options["googleapis"].shared = True
+            else:
+                self.options["protobuf/*"].shared = True
+                self.options["googleapis/*"].shared = True
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("protobuf/3.21.9", transitive_headers=True, run=can_run(self))
```
Comment (prince-chrismc) [inline recipes/grpc-proto/all/conanfile.py:57]:
```suggestion
       # protobuf symbols are exposed from generated structures 
       # https://github.com/conan-io/conan-center-index/pull/16185#issuecomment-1501174215 
        self.requires("protobuf/3.21.9", transitive_headers=True, transitive_libs=True, run=can_run(self))
```

**Comment by prince-chrismc** [comment]
@SpaceIm May i push my suggestion? I would like to unblock the google stuff 🙏

**Comment by prince-chrismc** [comment]
Literally ever PR I touch breaks 😭

**Comment by prince-chrismc** [comment]
v2 passed 🤞

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**